### PR TITLE
Suggested articles

### PR DESCRIPTION
**Pull request by UladBohdan**

### CODE REVIEW

File: components/articles/blocks/LatestArticles.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const LatestArticles = ({ block, data, blocks }) => {
   // TODO: to handle situation with multiple 'featured' blocks or 'featured' blocks
   // going *after* latestArticles.
   const { featured } = blocks;
-  let nextIndex = featured.frozen ? 0 : 1;
+  let nextIndex = featured && !featured.frozen ? 1 : 0;
```
Comment (Drapegnik) [inline components/articles/blocks/LatestArticles.js:19]: `featured?.frozen ? 0 : 1`

File: components/articles/compositions/ArticlesCompositionAsBlock.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  TopicsById,
+} from 'utils/customPropTypes';
+import { getArticlesBlocks } from 'utils/getters';
+
+import ArticlesComposition from './ArticlesComposition';
+
+const ArticlesCompositionAsBlock = ({ block, data }) => {
+  if (block.articles.length === 0) {
+    // It's ok to have block template in fibery: it should be ignored unless it is filled with data.
+    return <></>;
```
Comment (Drapegnik) [inline components/articles/compositions/ArticlesCompositionAsBlock.js]: `return null`

File: components/articles/compositions/ArticlesCompositionAsBlock.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  ArticlePreviewsById,
+  TagsById,
+  TopicsById,
+} from 'utils/customPropTypes';
+import { getArticlesBlocks } from 'utils/getters';
+
+import ArticlesComposition from './ArticlesComposition';
+
+const ArticlesCompositionAsBlock = ({ block, data }) => {
+  if (block.articles.length === 0) {
```
Comment (Drapegnik) [inline components/articles/compositions/ArticlesCompositionAsBlock.js]: `if (!articles.length)`

File: components/articles/compositions/ArticlesCompositionAsBlock.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  ArticlePreviewArray,
+  ArticlePreviewsById,
+  TagsById,
+  TopicsById,
+} from 'utils/customPropTypes';
+import { getArticlesBlocks } from 'utils/getters';
+
+import ArticlesComposition from './ArticlesComposition';
+
+const ArticlesCompositionAsBlock = ({ block, data }) => {
```
Comment (Drapegnik) [inline components/articles/compositions/ArticlesCompositionAsBlock.js]:
```suggestion
const ArticlesCompositionAsBlock = ({ block: {articles}, data }) => {
```

File: components/articles/compositions/ArticlesCompositionAsBlock.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  if (block.articles.length === 0) {
+    // It's ok to have block template in fibery: it should be ignored unless it is filled with data.
+    return <></>;
+  }
+  return (
+    <ArticlesComposition
+      articlesCount={block.articles.length}
+      blocks={getArticlesBlocks(block.articles.map(articleId => data.articles[articleId]))}
```
Comment (Drapegnik) [inline components/articles/compositions/ArticlesCompositionAsBlock.js]:
```suggestion
      blocks={getArticlesBlocks(articles.map(articleId => data.articles[articleId]))}
```

File: pages/index.js [hunk 6]
```diff
@@ -1,33 +1,26 @@
...
-    if (!Block) {
-      return null;
-    }
-    return (
-      // eslint-disable-next-line react/no-array-index-key
-      <Block key={index} block={block} data={data} blocks={blocksByType} lang={lang} />
-    );
-  });
-};
+const MainPage = ({ blocks, data }) => <CardsLayout blocks={blocks} data={data} />;
```
Comment (Drapegnik) [inline pages/index.js:23]: nice! short)

**Review by Drapegnik — CHANGES_REQUESTED** [review]
please fix small code style issues

File: components/articles/blocks/LatestArticles.js (hunk 1 above)
Comment (UladBohdan) [reply components/articles/blocks/LatestArticles.js:19]: it does not seem to be the same. in my case absense of `featured` is resolved as 0 while in your case absense of `featured` would make it 1

**Review by Drapegnik — APPROVED** [review]

**Comment by UladBohdan** [comment]
[Article Page: suggested articles block](https://trello.com/c/uiXGZKLB/433-article-page-suggested-articles-block)

**Pull request merged**